### PR TITLE
Added REST server manual

### LINKED ISSUE #451: Add REST server manual
Add a separate manual for the REST

### PR DESCRIPTION
**Pull request by JAmmermann-DLR**
First version of the server manual. The generated API doc is not yet included.

Closes #451

### CODE REVIEW

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+:imagesdir: images
+:title-logo-image: images/title/VirtualSatellite_Server.jpg
+:toc:
+:toclevels: 3
+:experimental: 
+
+== Introduction
+
+Virtual Satellite can be run headless as a server that provides a REST API. 
+The REST API can be used to access and manipulate the Virtual Satellite model without the user interface.
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:11]: I think this paragraph needs to be expanded a bit. I don't want us to explain in detail what REST is, but somebody who doesnt know what REST is (or just quickly googled it) shiould be able to read this. As a first step we could improve this by mentioning here that the API can be used to 'acess and manipulate the Virtual Satellite model _remotely_ and without the _desktop_ interface". I think that would make the purpose a bit clearer. But there's probably some more than can be done.

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+Between the API and the controller also is an _Access check_ layer.
+
+A REST Client can send a request to the provided _REST API_ that is partitioned into multiple Resources:
+An _Admin Resource_, in which an administrator can manipulate the Server Configuration (_ServerConf_), e.g. add Repository Configurations (_RepoConf_),
+and a _Repository Resource_. Each _Server Repository_ provided has its own _Subresource_.
+
+In the _Access check_ layer the credentials are obtained from the HTTP Header and it is checked if the requested Resources can be accessed.
+The credentials are handed over to be used for user role handling in the Virtual Satellite _Data Model_.
+
+The _Server Controller_ handles requests to the _Admin Resource_. In it's _ServerConf_ the server settings are stored.
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:62]: it's -> its

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+* The Management API for administrative tasks ({*name*} = *management*)
+
+The following documentation of the REST endpoints is automatically generated using Swagger.
+For each API the swagger documentation also is provided via the REST API at:
+[subs=+quotes]
+....
+{*server adress*}:8000/rest/{*name*}/{*version number*}/swagger.json or /swagger.yaml
+....
+
+It can help you, if you wan't to connect Virtual Satellite to your own application.
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:94]: wan't -> want

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]
Didn't really check much for typos. Only bigger problem is with the Introduction.

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc (hunk 1 above)
Comment (JAmmermann-DLR) [reply de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc]: Sure, I tried to rewrite it in a more user friendly way

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc [hunk 4]
```diff
@@ -0,0 +1,236 @@
+= Virtual Satellite 4 Core: Server Manual
+:imagesdir: images
+:title-logo-image: images/title/VirtualSatellite_Server.jpg
+:toc:
+:toclevels: 3
+:experimental: 
+
+== Introduction
+
+Virtual Satellite can be run headless as a server that allows you to connect Virtual Satellite to a network, e.g. the Internet.
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:10]: "Virtual Satellite can be run headless as a server that allows you to connect Virtual Satellite to a network, e.g. the Internet."
->
"Virtual Satellite can be run headless as a server.
that allows you to connect Virtual Satellite to a network, e.g. the Internet."

**Review by SaMuellerDLR — APPROVED** [review]
Great, just a minor sentence improvement.

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+A Server instance can maintain multiple _Server Repositories_, with each repository connected to a _git_ or _SVN repository_.
+_REST Clients_ can access the Server using the HTTP commands GET, PUT, DELETE and POST. 
+So a Client can read or write into the Virtual Satellite model. 
+The server provides functionality for administrators, that can manipulate the server configuration, and users.
+For that purpose it uses authentication and authorization.
+// For authentication purposes it is planned to an LDAP server can be connected to the REST server.
+// The connection can be secured via HTTPS.
+Every user can only access certain provided Server Repositories.
+
+The connected SVN/git repositories can also be directly changed from _Clients using the Virtual Satellite UI_.
```
Comment (franzTobiasDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:43]:
```suggestion
The connected SVN/git repositories can also be manipulated from _Clients using the Virtual Satellite UI_.
```
change sounded for me like you could change the remote or other git/svn properties; directly sounded like you could use UI to direktly manipulate the checked-out repo within the server...

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+For each API the swagger documentation also is provided via the REST API at:
+[subs=+quotes]
+....
+{*server adress*}:8000/rest/{*name*}/{*version number*}/swagger.json or /swagger.yaml
+....
+
+It can help you, if you want to connect Virtual Satellite to your own application.
+E.g. it can be used to automatically generate your client, using Swaggers Codegen capabilities.
+
+#TODO: insert generated swagger documentation for the APIs here#
```
Comment (franzTobiasDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:100]: Are you going to handle this TODO in this PR?

**Review by franzTobiasDLR — COMMENTED** [review]
Great work 👍

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc (hunk 6 above)
Comment (JAmmermann-DLR) [reply de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:100]: No, I want to adress it in https://github.com/virtualsatellite/VirtualSatellite4-Core/issues/834. But before that I would like to merge the two open swagger PRs.

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc (hunk 5 above)
Comment (JAmmermann-DLR) [reply de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:43]: Yeah I will change it

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+image::chapterServerOverview/REST_server_overview.png[Server Overview]
+
+A Server instance can maintain multiple _Server Repositories_, with each repository connected to a _git_ or _SVN repository_.
+_REST Clients_ can access the Server using the HTTP commands GET, PUT, DELETE and POST. 
+So a Client can read or write into the Virtual Satellite model. 
+The server provides functionality for administrators, that can manipulate the server configuration, and users.
+For that purpose it uses authentication and authorization.
+// For authentication purposes it is planned to an LDAP server can be connected to the REST server.
+// The connection can be secured via HTTPS.
+Every user can only access certain provided Server Repositories.
```
[the remark this answers is not in the document]
Comment (JAmmermann-DLR) [reply de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_Server_Manual.adoc:41]: So the idea is that not all user can access all repositories, but instead each authenticated user has certain repositories he can access (stored in the authentication file). I will think of a better way to say that, nevertheless I will probably add an authentication part to the docu once  it is activated (see https://github.com/virtualsatellite/VirtualSatellite4-Core/issues/812).

**Review by franzTobiasDLR — APPROVED** [review]

**Pull request merged**